### PR TITLE
Cost Surface - shapefile converting & extracting PU cost/id

### PR DESCRIPTION
**Pull request by kgajowy**
Includes moving `isDefined` into lib (`util`)

(yes, I messed up branch name with switching surface/cost order 😅 )

### CODE REVIEW

**Comment by kgajowy** [comment]
TODO: `e2e` specs don't see libs, config needs to be updated

File: api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.spec.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+describe(`when given GeoJson isn't a FeatureCollection`, () => {
+  it(`throws exception`, () => {
+    expect(() => sut.extract(fixtures.simpleGeometry())).toThrow(
+      /is supported/,
+    );
+  });
+});
+
+describe(`when given GeoJson has pu costs`, () => {
+  it(`throws exception`, () => {
```
Comment (hotzevzl) [inline api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.spec.ts:35]: the test description doesn't sound right

File: api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.spec.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    expect(() => sut.extract(fixtures.geoFeaturesWithoutCost())).toThrow(
+      /missing cost/,
+    );
+  });
+});
+
+describe(`when given GeoJson isn't a FeatureCollection`, () => {
+  it(`throws exception`, () => {
+    expect(() => sut.extract(fixtures.simpleGeometry())).toThrow(
+      /is supported/,
```
Comment (hotzevzl) [inline api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.spec.ts:29]: super minor - we'll see the test passing, but asserting that an exception will be raised containing the text `is supported` seems slightly misleading (why should an error be thrown if *something* "is supported"?): in this specific case, maybe matching case-insensitively the full text of the error (`/Only FeatureCollection is supported/i`) should make things clearer for whoever skims through these tests in the future.

File: api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { isDefined } from '@marxan/utils';
+import { FeatureCollection, GeoJSON, Geometry } from 'geojson';
+import { PlanningUnitCost } from '../ports/planning-unit-cost';
+import { PuExtractorPort } from '../ports/pu-extractor/pu-extractor.port';
+
+type Properties = {
+  cost: number;
+  planningUnitId: string;
+};
+type MaybeProperties = Partial<Properties> | undefined | null;
```
Comment (hotzevzl) [inline api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.ts:10]: this is quite useful in general - I suggest to create a `MaybeSome<T>` and then `type MaybeProperties = MaybeSome<Properties>` or something like that.

File: api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+export class ExtractPuCost implements PuExtractorPort {
```
Comment (hotzevzl) [inline api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.ts:12]: I am not too keen on class names based on verbs - why don't we name this `PuCostExtractor`?

Comment (hotzevzl) [reply api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.ts:12]: I think this would also match some of your previous code

**Review by hotzevzl — APPROVED** [review]

File: api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.ts (hunk 3 above)
Comment (kgajowy) [reply api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.ts]: https://github.com/Vizzuality/marxan-cloud/pull/219/commits/573622f92a11fdd126982184dc16a4ab3e03ecb3

File: api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.ts (hunk 4 above)
Comment (kgajowy) [reply api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.ts]: https://github.com/Vizzuality/marxan-cloud/pull/219/commits/e9ae13e05dc6fddc7d10fb612c7b378de4b495af

File: api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.spec.ts (hunk 2 above)
Comment (kgajowy) [reply api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.spec.ts]: fixed in https://github.com/Vizzuality/marxan-cloud/pull/219/commits/573622f92a11fdd126982184dc16a4ab3e03ecb3

File: api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.spec.ts (hunk 1 above)
Comment (kgajowy) [reply api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.spec.ts]: fixed in https://github.com/Vizzuality/marxan-cloud/pull/219/commits/573622f92a11fdd126982184dc16a4ab3e03ecb3

File: api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.ts (hunk 3 above)
Comment (Dyostiq) [reply api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.ts]: https://github.com/piotrwitek/utility-types#nullish

**Pull request merged**